### ISSUE TITLE
Is it possible to have SDL3's D3D11 renderer and Direct2D interoperability?

### ISSUE THREAD

**Issue by STVHA** [libsdl-org/SDL#9775]
Currently my test using SDL3 v3.1.2 is working well via D3D11 renderer (except a little memleak), and I want to create a Direct2D from SDL's D3D11 that I retrieved via SDL_PROP_RENDERER_D3D11_DEVICE_POINTER. I, however, could find any way to make the interoperability work because there is no any API function to access D3D11's swapchain or its back buffer.
Is there a walk around for that or should you add one more property like SDL_PROP_RENDERER_D3D11_SWAPCHAIN to return the internal swapChain?

**Comment by slouken** [comment]
Good point. I added a property for the swapchain to the D3D11 and D3D12 renderers. Thanks!

**Comment by STVHA** [comment]
Thank you @slouken for quickly appending the SDL properties.
My experiment of drawing D2D over D3D11 with the latest main commit works pretty well.

However I found an issue relating to window resizing. That is I used "testffmpeg" for the D2D test, the app sometimes raises error exception when I resize the video window. To avoid confusion I have disabled the D2D's relevant things but the error still persists.
The point in testffmpeg code that raises the error: 
```
avcodec_send_packet(video_context, pkt);
```
The function in ffmpeg code the debugger stops at:
```
int ff_dxva2_common_end_frame(..) {  
 --> ID3D11VideoContext_DecoderBeginFrame(..);
}
```
Could you take a look to testffmpeg to see if any possibility of the window resizing issue?

**Comment by slouken** [comment]
What is the actual error? Is it a fatal error, or an exception that's handled internally?

**Comment by STVHA** [comment]
It is an unhandled exception error something like below:
`Exception thrown at 0x00007FFEB8DE53AC in accelsdraw.exe: Microsoft C++ exception: _com_error at memory location 0x0000007FBF54EC38.`
The code of "accelsdraw" is almost same as testffmpeg's except the D2D part.
As I mentioned previously the exception raises sometimes only when I resize the window.

**Comment by slouken** [comment]
Try continuing. I've seen exceptions like these before, and they're usually not fatal. If you enable the D3D11 debug layer with SDL_HINT_RENDER_DIRECT3D11_DEBUG you might get some more details.

**Comment by STVHA** [comment]
There is no difference with whatever state of SDL_HINT_RENDER_DIRECT3D11_DEBUG, even with SDL_HINT_RENDER_DIRECT3D_THREADSAFE.

Something more I found:
- I found a way to trigger the symptom: set a breaking point (MSVS 2022) at any line inside the loop of decoding/rendering, take some steps, then remove the breakpoint and hit F5 to continue, the symptom should happen immediately exactly same as if it happends like resizing window. I have a pic below.
![SDL3-Exception](https://github.com/libsdl-org/SDL/assets/142473182/d70108e5-bc52-43e4-bd84-13b931112d2c)

- If I setup the ffmpeg decoder to use CPU instead of D3D11, the symptom seem not occur anymore. 

So in my guess there may be an internal synchronization issue between D3D11 decoder and D3D11 renderer.
